azurerm/internal/services/apimanagement/api_management_subscription_resource.go

```diff
@@ -150,7 +150,7 @@ func resourceArmApiManagementSubscriptionCreateUpdate(d *schema.ResourceData, me
 	}
 
 	sendEmail := utils.Bool(false)
-	_, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, subscriptionId, params, sendEmail, "")
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, subscriptionId, params, sendEmail, "", "")
```

Comment:
what's this empty value here? as per the last PR this is expecting where this resource is being created from, so sending the Old/New Portal is misleading - presumably we should be sending `terraform` here?

Reply:
I will have a try

Reply:
I don't believe passing `terraform` is an option, the documentation says: `Default is old publisher portal. Possible values include: 'developerPortal'`

```
UserCreateParameters.AppType Property

Namespace: Microsoft.Azure.Management.ApiManagement.Models
Assembly: Microsoft.Azure.Management.ApiManagement.dll
Package: Microsoft.Azure.Management.ApiManagement v6.0.0-preview

Gets or sets determines the type of application which send the create user request. Default is old publisher portal. Possible values include: 'developerPortal'
```

Reply:
Yeah that's where I was going with this - it's an implementation detail, passing old or new portal is misleading (since we're not provisioning from either of those, presumably this is used in emails that get sent?) - so the API either needs to support other or this field likely not configurable?

It's not necessarily a blocker, but it's misleading when provisioning resources which would cause confusion?

Reply:
Totally agree, can we merge this to unblock all the PR's and I can follow up with the service team on Redmond time Monday?

Reply:
@ArcturusZhang how did you get on with testing this out, did the API accept this value?

Reply:
@tombuildsstuff @ArcturusZhang I have heard back from the service team:

>From the [release notes](https://azure.microsoft.com/en-us/updates/azure-api-management-update-july-2020/):
>
The PUT Subscription API call now accepts the `appType` query parameter in API versions 2018-06-01-preview or later. The value of the query parameter controls destination of the link included in the subscription confirmation email sent to a user - portal (default) results in a link pointing to the `legacy developer portal`, while `developerPortal` results in a link pointing to the new developer portal.
>
>1.	I suggest that the default in Terraform is `developerPortal` (pointing to new portal)...
>2.	There are no plans to make it a free-form string and there shouldn’t be `other` because we have only two portals – `legacy developer portal` and `new developer portal`. 

So per the service team we should us `developerPortal` instead of the default `portal`, seems rather odd to me that they would make the `legacy portal` the default instead of the `developerPortal`, but I digress... 🙃

Reply:
🤷 seems fine for now